imports/plugins/core/ui/client/containers/buttonSelectContainer.js

```diff
@@ -0,0 +1,133 @@
+import React, { Component, PropTypes } from "react";
+import classnames from "classnames";
+import { composeWithTracker, merge } from "/lib/api/compose";
+import Button from "../components/button/button.jsx";
+import { ButtonSelect } from "../components/button";
+
+class ButtonSelectContainer extends Component {
```

Comment:
We have two sets of components in Reaction.

`Container` components and `Presentational` components.

A Presentational component is like the `Select` or `Button` component, which **do not** connect to an external datasource though `composeWithTracker`, `redux`, or some other method. It's a simple component that can be used anywhere and for any purpose and any manipulation of that view is usually done from a parent or `Container` component through props (avoiding `state` unless absolutely necessary).

`Container` components contain all the logic for a single component. The `Container` component usually wraps a `Presentational` component and is to be used when you want a dumb `Presentational` component to do more with data from external sources. For example the `ProductDetailContainer` fetches product, variants, and media and passes them to the `ProductDetail` `Presentational` component. All the `ProductDetail` component does is render the view.

This is good practice in React so you force yourself to make reusable dumb components, and move the control of them higher up in the render tree to a container component / redux store.

**tl;dr**

Container components = business logic, data fetching, data manipulation
View components = accept props, and render a view based on those props